File: nilon-invoices/src/renderer/services/authService.ts
import { UserRole } from '../stores/authStore';

export interface AuthResponse {
  success: boolean;
  user?: {
    username: string;
    role: UserRole;
  };
  error?: string;
}

export const authService = {
  login: async (username: string, password: string): Promise<AuthResponse> => {
    // Artificial delay to simulate network/IPC response
    await new Promise((resolve) => setTimeout(resolve, 800));

    // Hard-coded authentication for default Admin account
    if (username === 'Admin' && password === '123456') {
      return { success: true, user: { username: 'Admin', role: 'admin' } };
    }

    // Future implementation:
    // try {
    //   const response = await window.ipcRenderer.invoke('auth:login', { username, password });
    //   return response;
    // } catch (err) {
    //   return { success: false, error: 'Database connection failed' };
    // }

    // Hard-coded fallback for an arbitrary staff account if needed, or simply fail
    if (username === 'Staff' && password === '123456') {
      return { success: true, user: { username: 'Staff', role: 'staff' } };
    }

    return { success: false, error: 'Sai tài khoản hoặc mật khẩu.' };
  }
};

File: nilon-invoices-app/lib/services/auth_service.dart
import '../models/user_model.dart';

class AuthResult {
  final bool success;
  final UserModel? user;
  final String? error;

  AuthResult({
    required this.success,
    this.user,
    this.error,
  });
}

class AuthService {
  static Future<AuthResult> login(String username, String password) async {
    // Artificial delay
    await Future.delayed(const Duration(milliseconds: 600));

    if (username.trim() == 'Admin' && password == '123456') {
      return AuthResult(
        success: true,
        user: UserModel(username: 'Admin', role: UserRole.admin),
      );
    }

    if (username.trim() == 'Staff' && password == '123456') {
      return AuthResult(
        success: false,
        user: UserModel(username: 'Staff', role: UserRole.staff),
        error: 'Từ chối truy cập: Tài khoản Nhân viên không có quyền truy cập. Ứng dụng này chỉ dành riêng cho Admin.',
      );
    }

    return AuthResult(
      success: false,
      error: 'Sai tên tài khoản hoặc mật khẩu. Vui lòng kiểm tra lại.',
    );
  }
}
